packages/ibm-products/src/components/Datagrid/Datagrid/DatagridVirtualBody.tsx

```diff
@@ -85,6 +87,17 @@ const DatagridVirtualBody = (datagridState: DataGridState) => {
       if (headWrapEl) {
         headWrapEl.scrollLeft = virtualBody?.scrollLeft;
       }
+      if (enableSpacerColumn === true) {
+        const spacerColumn: HTMLDivElement | null = document.querySelector(
```

Comment:
We might want to add an optional chain here to be sure to avoid any SSR issues, like above. Actually, you could use the element above `headWrapEl` and do `headWrapEl.querySelector('thead th:last-child')` since you already have that element.